EC2 updates

pmhahn opened the pull request:
The EC2 Instance Type information contains some errors (especially the disk size for the m3 instances), conversion errors between GiB an M[i]B, disk count, ..
Also g2.8xlarge and t2.large are missing.

pmhahn committed:
EC2: Specify RAM size in GiB

The values on <https://aws.amazon.com/ec2/instance-types/?nc1=f_ls> are
given in [GiB].

Let Python convert them to [MiB], as the current values are a mix of
[MB] and [MiB].

pmhahn committed:
EC2: Fix disk size

The values on <https://aws.amazon.com/ec2/instance-types/?nc1=f_ls> are
given in [GB].

Let Python calculate the size of multiple disks.

Remove wrong factor 1000 from m3 instances.

pmhahn committed:
EC2: Add missing vCPU information

Add vCPU information from
<https://aws.amazon.com/ec2/instance-types/?nc1=f_ls>

pmhahn committed:
EC2: Add g2.8xlarge and t2.large

From <https://aws.amazon.com/ec2/instance-types/?nc1=f_ls>

pmhahn committed:
EC2: Rename i2 and d2 instance types

i2 are I/O optimized, while d2 are dense optimized

pmhahn committed:
EC2: Move comment before t2 instances

t2.micro is also a "Burstable Performance General Purpose" instance
type.

pmhahn committed:
EC2: Add clarifications

Add source URLs.
Add unit type for RAM and disk size.

tonybaloney commented:
@pmhahn thanks for the submission,  the tests are failing because pricing.json is missing the new instance types, e.g.
KeyError: 'g2.8xlarge'
Please can you add the missing prices and resubmit

pmhahn committed:
EC2: Update pricing

Please note that the scraper stores the price as string and does not
beautify the string to have exactly 3 digits in the radix part.

pmhahn committed:
Remove duplicate pricing info

Commit b7d851f1 added the pricing info twice.

pmhahn committed:
EC2: Sort pricing by (non-)numeric part

contrib/scrape-ec2-prices.py changes the ordering of data/pricing.json,
as it sorts the dictionary keys by string, which puts '10' before '2'.

Add a function to split the dictionary keys into the numeric, alphabetic
and other part and compare the first as integer, the middle by meaning,
and the later as string.
